Set path to MPAS model relative to root of git branch

xylar opened the pull request:
This requires storing the root of the git branch in an environment variable.

closes #361

xylar commented:
@matthewhoffman, I think this will work but I'm still testing.

xylar commented:
@matthewhoffman, I believe this will do the trick.  Could you try it out when you have a chance?

matthewhoffman commented:
@xylar , thanks for throwing this together so quickly.  Strangely, when I try it, the tests set up fine from any directory, but when I try to run it, they all fly past in about 1 second and say they all pass, but clearly have not actually run:
```
landice/dome/2000m/sia_restart_test
  test execution:      SUCCESS
landice/dome/2000m/sia_decomposition_test
  test execution:      SUCCESS
landice/dome/variable_resolution/sia_restart_test
  test execution:      SUCCESS
landice/dome/variable_resolution/sia_decomposition_test
  test execution:      SUCCESS
landice/enthalpy_benchmark/A
  test execution:      SUCCESS
landice/eismint2/decomposition_test
  test execution:      SUCCESS
landice/eismint2/enthalpy_decomposition_test
  test execution:      SUCCESS
landice/eismint2/restart_test
  test execution:      SUCCESS
landice/eismint2/enthalpy_restart_test
  test execution:      SUCCESS
landice/greenland/sia_restart_test
  test execution:      SUCCESS
landice/greenland/sia_decomposition_test
  test execution:      SUCCESS
landice/hydro_radial/restart_test
  test execution:      SUCCESS
landice/hydro_radial/decomposition_test
  test execution:      SUCCESS
landice/dome/2000m/fo_decomposition_test
  test execution:      SUCCESS
landice/dome/2000m/fo_restart_test
  test execution:      SUCCESS
landice/dome/variable_resolution/fo_decomposition_test
  test execution:      SUCCESS
landice/dome/variable_resolution/fo_restart_test
  test execution:      SUCCESS
landice/circular_shelf/decomposition_test
  test execution:      SUCCESS
landice/greenland/fo_decomposition_test
  test execution:      SUCCESS
landice/greenland/fo_restart_test
  test execution:      SUCCESS
landice/thwaites/decomposition_test
  test execution:      SUCCESS
landice/thwaites/restart_test
  test execution:      SUCCESS
Test Runtimes:
00:00 PASS landice_dome_2000m_sia_restart_test
00:00 PASS landice_dome_2000m_sia_decomposition_test
00:00 PASS landice_dome_variable_resolution_sia_restart_test
00:00 PASS landice_dome_variable_resolution_sia_decomposition_test
00:00 PASS landice_enthalpy_benchmark_A
00:00 PASS landice_eismint2_decomposition_test
00:00 PASS landice_eismint2_enthalpy_decomposition_test
00:00 PASS landice_eismint2_restart_test
00:00 PASS landice_eismint2_enthalpy_restart_test
00:00 PASS landice_greenland_sia_restart_test
00:00 PASS landice_greenland_sia_decomposition_test
00:00 PASS landice_hydro_radial_restart_test
00:00 PASS landice_hydro_radial_decomposition_test
00:00 PASS landice_dome_2000m_fo_decomposition_test
00:00 PASS landice_dome_2000m_fo_restart_test
00:00 PASS landice_dome_variable_resolution_fo_decomposition_test
00:00 PASS landice_dome_variable_resolution_fo_restart_test
00:00 PASS landice_circular_shelf_decomposition_test
00:00 PASS landice_greenland_fo_decomposition_test
00:00 PASS landice_greenland_fo_restart_test
00:00 PASS landice_thwaites_decomposition_test
00:00 PASS landice_thwaites_restart_test
Total runtime 00:01
PASS: All passed successfully!
```

If I look at one of the case_outputs, it looks incomplete, but there are no errors:
```
compass calling: compass.landice.tests.dome.restart_test.RestartTest.run()
  inherited from: compass.testcase.TestCase.run()
  in /turquoise/usr/projects/climate/mhoffman/compass/compass/compass/testcase.py

Running steps: 

compass calling: compass.landice.tests.dome.restart_test.RestartTest.validate()
  in /turquoise/usr/projects/climate/mhoffman/compass/compass/compass/landice/tests/dome/restart_test/__init__.py

Skipping validation because full_run and restart_run weren't  run
```

xylar commented:
Sorry about that, @matthewhoffman.  This took me awhile to figure out.

It turns out that a combination of things caused this.  I was setting a config option in `setup.py` to add `compass_branch` as a config option.  But I was doing this before adding other config files, making `setup.py` the lowest priority config file in the meta-config parser.  Later, when `steps_to_run` got set, it was overridden by the version from `default.cfg`, which is empty.

I have fixed this in 2 ways.  First, I removed `steps_to_run` from default.cfg because it isn't useful or needed there.  I think it was just for commenting, which we should handle differently.  Second, I moved the place where `compass_branch` gets added to much later.

Anyway, I don't expect this kind of thing to be a problem elsewhere but it was a surprise in `setup.py`.

Let me know if this works for you now.  It seems to be running the `nightly` ocean suite on my laptop just fine now.

matthewhoffman commented:
Thanks for the update @xylar.  I'll test this again in the same way once Badger is back online.

matthewhoffman reviewed (approved):
@xylar , I tested again and it works great with the changes you made!  I ran `compass suite` from my home directory and then ran the test suite on a compute node successfully.

xylar commented:
Great, thanks @matthewhoffman.  Glad we got this fixed for you!
